--- mingwin32.mak.orig	2014-06-06 18:33:43.768283300 +0200
+++ mingwin32.mak	2014-06-06 18:48:21.819028600 +0200
@@ -30,8 +30,8 @@
 
 CFLAGS += -I$(PDCURSES_SRCDIR)
 
-BASEDEF		= $(PDCURSES_SRCDIR)\exp-base.def
-WIDEDEF		= $(PDCURSES_SRCDIR)\exp-wide.def
+BASEDEF		= $(PDCURSES_SRCDIR)/exp-base.def
+WIDEDEF		= $(PDCURSES_SRCDIR)/exp-wide.def
 
 DEFDEPS		= $(BASEDEF)
 
@@ -44,23 +44,29 @@
 	CFLAGS += -DPDC_FORCE_UTF8
 endif
 
-DEFFILE		= pdcurses.def
+ifeq ($(UTF8),Y)
+	LIBNAME = libpdcursesu
+else ifeq ($(WIDE),Y)
+	LIBNAME = libpdcursesw
+else
+	LIBNAME = libpdcurses
+endif
+
+DEFFILE		= $(LIBNAME).def
 
 LINK		= gcc
 
 ifeq ($(DLL),Y)
 	CFLAGS += -DPDC_DLL_BUILD
 	LIBEXE = gcc $(DEFFILE)
-	LIBFLAGS = -Wl,--out-implib,pdcurses.a -shared -o
-	LIBCURSES = pdcurses.dll
+	LIBCURSES = $(LIBNAME).dll
+	LIBFLAGS = -Wl,--out-implib,$(LIBCURSES).a -shared -o
 	LIBDEPS = $(LIBOBJS) $(PDCOBJS) $(DEFFILE)
-	CLEAN = $(LIBCURSES) *.a $(DEFFILE)
 else
 	LIBEXE = ar
 	LIBFLAGS = rcv
-	LIBCURSES = pdcurses.a
+	LIBCURSES = $(LIBNAME).a
 	LIBDEPS = $(LIBOBJS) $(PDCOBJS)
-	CLEAN = *.a
 endif
 
 .PHONY: all libs clean demos dist
@@ -70,24 +76,26 @@
 libs:	$(LIBCURSES)
 
 clean:
-	-del *.o
-	-del *.exe
-	-del $(CLEAN)
+	-rm -f *.o
+	-rm -f *.exe
+	-rm -f *.a
+	-rm -f *.dll
+	-rm -f *.def
 
 demos:	$(DEMOS)
 	strip *.exe
 
 $(DEFFILE): $(DEFDEPS)
-	echo LIBRARY pdcurses > $@
+	echo LIBRARY $(LIBNAME) > $@
 	echo EXPORTS >> $@
-	type $(BASEDEF) >> $@
+	cat $(BASEDEF) >> $@
 ifeq ($(WIDE),Y)
-	type $(WIDEDEF) >> $@
+	cat $(WIDEDEF) >> $@
 endif
 
 $(LIBCURSES) : $(LIBDEPS)
 	$(LIBEXE) $(LIBFLAGS) $@ $?
-	-copy pdcurses.a panel.a
+	-cp $(LIBCURSES).a panel.a
 
 $(LIBOBJS) $(PDCOBJS) : $(PDCURSES_HEADERS)
 $(PDCOBJS) : $(PDCURSES_WIN_H)
