diff -uNr SDL2-2.0.3.orig/src/haptic/windows/SDL_syshaptic.c SDL2-2.0.3.modif/src/haptic/windows/SDL_syshaptic.c
--- SDL2-2.0.3.orig/src/haptic/windows/SDL_syshaptic.c	2014-03-16 03:31:41.000000000 +0100
+++ SDL2-2.0.3.modif/src/haptic/windows/SDL_syshaptic.c	2015-05-12 21:23:52.518581000 +0200
@@ -960,6 +960,10 @@
         SDL_free(item);
     }
 
+    numhaptics = 0;
+    SDL_hapticlist = NULL;
+    SDL_hapticlist_tail = NULL;
+
     if (loaded_xinput) {
         WIN_UnloadXInputDLL();
         loaded_xinput = SDL_FALSE;
