diff -uNr SDL2-2.0.4.orig/src/haptic/windows/SDL_dinputhaptic.c SDL2-2.0.4.mod/src/haptic/windows/SDL_dinputhaptic.c
--- SDL2-2.0.4.orig/src/haptic/windows/SDL_dinputhaptic.c	2016-10-30 09:04:14.539315700 +0100
+++ SDL2-2.0.4.mod/src/haptic/windows/SDL_dinputhaptic.c	2016-10-30 11:09:55.475158300 +0100
@@ -1027,6 +1027,19 @@
     /* Create the actual effect. */
     ret =
         IDirectInputEffect_SetParameters(effect->hweffect->ref, &temp, flags);
+    if (ret == DIERR_NOTEXCLUSIVEACQUIRED) {
+        IDirectInputDevice8_Unacquire(haptic->hwdata->device);
+        ret = IDirectInputDevice8_SetCooperativeLevel(haptic->hwdata->device, SDL_HelperWindow, DISCL_EXCLUSIVE | DISCL_BACKGROUND);
+        if (SUCCEEDED(ret)) {
+            ret = DIERR_NOTACQUIRED;
+        }
+    }
+    if (ret == DIERR_INPUTLOST || ret == DIERR_NOTACQUIRED) {
+        ret = IDirectInputDevice8_Acquire(haptic->hwdata->device);
+        if (SUCCEEDED(ret)) {
+            ret = IDirectInputEffect_SetParameters(effect->hweffect->ref, &temp, flags);
+        }
+    }
     if (FAILED(ret)) {
         DI_SetError("Unable to update effect", ret);
         goto err_update;
