diff -uNr SDL2-2.0.4.orig/src/haptic/windows/SDL_dinputhaptic.c SDL2-2.0.4.mod/src/haptic/windows/SDL_dinputhaptic.c
--- SDL2-2.0.4.orig/src/haptic/windows/SDL_dinputhaptic.c	2016-10-02 19:49:40.955632500 +0200
+++ SDL2-2.0.4.mod/src/haptic/windows/SDL_dinputhaptic.c	2016-10-02 19:49:54.368868600 +0200
@@ -331,15 +331,6 @@
             goto acquire_err;
         }
 
-        /* Get number of axes. */
-        ret = IDirectInputDevice8_EnumObjects(haptic->hwdata->device,
-                                              DI_DeviceObjectCallback,
-                                              haptic, DIDFT_AXIS);
-        if (FAILED(ret)) {
-            DI_SetError("Getting device axes", ret);
-            goto acquire_err;
-        }
-
         /* Acquire the device. */
         ret = IDirectInputDevice8_Acquire(haptic->hwdata->device);
         if (FAILED(ret)) {
@@ -348,6 +339,15 @@
         }
     }
 
+    /* Get number of axes. */
+    ret = IDirectInputDevice8_EnumObjects(haptic->hwdata->device,
+                                          DI_DeviceObjectCallback,
+                                          haptic, DIDFT_AXIS);
+    if (FAILED(ret)) {
+        DI_SetError("Getting device axes", ret);
+        goto acquire_err;
+    }
+
     /* Reset all actuators - just in case. */
     ret = IDirectInputDevice8_SendForceFeedbackCommand(haptic->hwdata->device,
                                                        DISFFC_RESET);
